llm_applications_library/llm/generators/schema.py

```
@@ -145,6 +182,13 @@ def _alias_max_tokens(cls, data):
                 data["max_output_tokens"] = data.pop("max_tokens")
         return data
 
+    @model_validator(mode="after")
+    def _validate_logprobs(self):
+        """Validate that top_logprobs requires logprobs=True."""
+        if self.top_logprobs is not None and not self.logprobs:
+            raise ValueError("top_logprobs requires logprobs=True")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)